aspnetcore/signalr/hubs/sample/Hubs/ChatHub.cs

```diff
@@ -0,0 +1,42 @@
+﻿using Microsoft.AspNetCore.SignalR;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace SignalRChat
+{
+    public class ChatHub : Hub
+    {
+        public Task SendMessageToAll(string user, string message)
+        {
+            var timestamp = DateTime.Now.ToShortDateString();
+            return Clients.All.SendAsync("ReceiveMessage", timestamp, user, message);           
+        }
+        public Task SendMessageToMultipleConnections(string user, string message, List<string> connections)
+        {
+            var timestamp = DateTime.Now.ToShortDateString();
+            return Clients.Clients(connections.ToArray()).SendAsync("ReceiveMessage", timestamp, user, message);
+        }
+        public Task SendMessageToSingleConnection(string user, string message, string connection)
+        {
+            var timestamp = DateTime.Now.ToShortDateString();
+            return Clients.Client(connection).SendAsync("ReceiveMessage", timestamp, user, message, connection);
+        }
+        public Task SendMessageToGroup(string user, string message, string group)
+        {
+            var timestamp = DateTime.Now.ToShortDateString();
+            return Clients.Group(group).SendAsync("ReceiveMessage", timestamp, user, message, group);
+        }
+
+
+        private string _connectionID;
```

Comment:
As @anurse was saying, we don't want to talk about connection Id too much. Also this code is wrong, you get a new instance of a hub every method invocation so once `OnConnectedAsync` returns, `_connectionID` will be empty next time a function is called